cmake/kconfig.cmake

```diff
@@ -18,28 +18,14 @@ set(ENV{KERNELVERSION}      ${PROJECT_VERSION})
 set(ENV{KCONFIG_CONFIG}     ${DOTCONFIG})
 set(ENV{KCONFIG_AUTOHEADER} ${AUTOCONF_H})
 
-set(kconfig_target_list
-  config
-  gconfig
-  menuconfig
-  pymenuconfig
-  oldconfig
-  xconfig
-  )
-
-set(COMMAND_FOR_config       ${KCONFIG_CONF} --oldaskconfig ${KCONFIG_ROOT})
-set(COMMAND_FOR_gconfig      gconf                          ${KCONFIG_ROOT})
-set(COMMAND_FOR_menuconfig   ${KCONFIG_MCONF}               ${KCONFIG_ROOT})
-set(COMMAND_FOR_pymenuconfig ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/kconfig/menuconfig.py ${KCONFIG_ROOT})
-set(COMMAND_FOR_oldconfig    ${KCONFIG_CONF} --oldconfig    ${KCONFIG_ROOT})
-set(COMMAND_FOR_xconfig      qconf                          ${KCONFIG_ROOT})
-
-set(COMMAND_RUNS_ON_WIN_pymenuconfig 1)
+set(kconfig_target_list menuconfig)
+set(COMMAND_FOR_menuconfig ${PYTHON_EXECUTABLE} ${ZEPHYR_BASE}/scripts/kconfig/menuconfig.py ${KCONFIG_ROOT})
+set(COMMAND_RUNS_ON_WIN_menuconfig 1)
```

Comment:
We don't need this construct (RUNS_ON_WIN) now that everything runs on all platforms.

Reply:
We don't need it in `kconfig.cmake` but it still makes sense in other places, like for example QEMU?

Reply:
Yeah... `kconfig_target_list` is a bit silly now too.

Reply:
Correct, but this construct is not used in QEMU. So it is safe to remove it.

Reply:
Agreed, let's get rid of it

Reply:
Got rid of it.